### PR TITLE
Fix for #948: Improve target destruction reliability

### PR DESCRIPTION
**Pull request by ggeurts**

### CODE REVIEW

File: lib/PuppeteerSharp/ChromiumProcess.cs [hunk 1]
```diff
@@ -193,7 +193,7 @@ public async Task<bool> WaitForExitAsync(TimeSpan? timeout)
                 await _exitCompletionSource.Task.WithTimeout(() =>
                 {
                     taskCompleted = false;
-                }, timeout.Value.Milliseconds).ConfigureAwait(false);
+                }, ToTimeoutInt32(timeout.Value)).ConfigureAwait(false);
```
Comment (kblok) [inline lib/PuppeteerSharp/ChromiumProcess.cs]: Why do we need this?

Comment (ggeurts) [reply lib/PuppeteerSharp/ChromiumProcess.cs]: `timeout.Value.Milliseconds` only returns the milliseconds portion of the TimeSpan. Instead TotalMilliseconds should be used, but that may result in a value greater than `int.MaxValue`.

Comment (kblok) [reply lib/PuppeteerSharp/ChromiumProcess.cs]: Maybe we can improve `WithTimeout`, adding an overload with a `TimeSpan` parameter (which would be the normal flow internally)

Comment (ggeurts) [reply lib/PuppeteerSharp/ChromiumProcess.cs]: `TimeSpan.Milliseconds` returns only the milliseconds portion of the `TimeSpan`. Instead, `TotalMilliseconds` should be used, but that returns a `double` that mat be greater than `int.MaxValue`.

**Comment by kblok** [comment]
@ggeurts I think we need both. In many places are sending milliseconds from option arguments.

File: lib/PuppeteerSharp/Helpers/TaskHelper.cs [hunk 2]
```diff
@@ -9,22 +9,33 @@ namespace PuppeteerSharp.Helpers
     /// </summary>
     public static class TaskHelper
     {
+        private static readonly Func<int, Exception> DefaultExceptionFactory =
```
Comment (kblok) [inline lib/PuppeteerSharp/Helpers/TaskHelper.cs]: How about if we make it TimeSpan by default?
The methods expecting milliseconds can use TimeSpan.FromMilliseconds(int)
And we wouldn't need a `ToTimeoutInt32`

Comment (ggeurts) [reply lib/PuppeteerSharp/Helpers/TaskHelper.cs]: True, and I considered it, but `CancellationTokenSource` will convert the `TimeSpan` back to `int` and also throw a `ArgumentOutOfRangeException`.

Comment (kblok) [reply lib/PuppeteerSharp/Helpers/TaskHelper.cs]: Genius!

**Pull request merged**